### PR TITLE
Feature/vih 5515 task 5489 get chat messages

### PR DESCRIPTION
**Pull request by jdssunil**
### JIRA link (if applicable) ###
Task : https://tools.hmcts.net/jira/browse/VIH-5488
Task : https://tools.hmcts.net/jira/browse/VIH-5489
Main Story : https://tools.hmcts.net/jira/browse/VIH-5515

### Change description ###
Created a new table 'Message' to store instant messages
Query commands 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: VideoAPI/VideoApi.Contract/Responses/MessageResponse.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace VideoApi.Contract.Responses
+{
+    public class MessageResponse
+    {
+        public long Id { get; set; }
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Contract/Responses/MessageResponse.cs]: we do not need to expose ID

File: VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetMessagesForConferenceQueryTests.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+using System.Linq;
+using Faker;
+using FluentAssertions;
+using NUnit.Framework;
+using Testing.Common.Helper.Builders.Domain;
+using VideoApi.DAL;
+using VideoApi.DAL.Exceptions;
+using VideoApi.DAL.Queries;
+using VideoApi.Domain.Enums;
+using Task = VideoApi.Domain.Task;
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetMessagesForConferenceQueryTests.cs]: why are you using a task? this shouldn't be neeed for this test

File: VideoAPI/VideoApi.Contract/Responses/ConferenceDetailsResponse.cs [hunk 3]
```diff
@@ -65,6 +65,11 @@ public class ConferenceDetailsResponse
         public MeetingRoomResponse MeetingRoom { get; set; }
         
         public string HearingVenueName { get; set;  }
-        
+
+        /// <summary>
+        /// List of participants in conference
```
Comment (rehmanab) [inline VideoAPI/VideoApi.Contract/Responses/ConferenceDetailsResponse.cs]: Whats this

Comment (jdssunil) [reply VideoAPI/VideoApi.Contract/Responses/ConferenceDetailsResponse.cs]: Copy and paste error

File: VideoAPI/VideoApi.Domain/Conference.cs [hunk 4]
```diff
@@ -175,5 +177,15 @@ public Participant GetVideoHearingOfficer()
         {
             return Participants.SingleOrDefault(x => x.IsVideoHearingOfficer());
         }
+
+        public IList<Message> GetMessages()
+        {
+            return Messages.OrderByDescending(x => x.TimeStamp).ToList();
+        }
+        public void AddMessage(string from, string to, string messageText)
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Domain/Conference.cs]: I can't seem to find unit tests for these methods. Look under the Domain => Conference folder in Unit Tests project for examples.

'AddParticipantTests'

File: VideoAPI/VideoApi.Domain/Conference.cs [hunk 5]
```diff
@@ -175,5 +177,15 @@ public Participant GetVideoHearingOfficer()
...
+        {
+            var message = new Message(from, to, messageText);
+            Messages.Add(message);
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Domain/Conference.cs]: There should be a clear messages function too (including tests)

File: VideoAPI/VideoApi.DAL/Migrations/20200205131422_AddMessage.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public partial class AddMessage : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "Message",
+                columns: table => new
+                {
+                    Id = table.Column<long>(nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
```
Comment (RuwanikaWeerakkody) [inline VideoAPI/VideoApi.DAL/Migrations/20200205131422_AddMessage.cs]: Hi Sunil, I don't think any of these fields should allow null

From = table.Column(nullable: true),
To = table.Column(nullable: true),
MessageText = table.Column(nullable: true),
TimeStamp = table.Column(nullable: false),
ConferenceId = table.Column(nullable: true)

Comment (jdssunil) [reply VideoAPI/VideoApi.DAL/Migrations/20200205131422_AddMessage.cs]: Done

File: VideoAPI/VideoApi.Domain/Conference.cs (hunk 5 above)
Comment (jdssunil) [reply VideoAPI/VideoApi.Domain/Conference.cs]: Done

File: VideoAPI/VideoApi.Domain/Conference.cs (hunk 4 above)
Comment (jdssunil) [reply VideoAPI/VideoApi.Domain/Conference.cs]: Done

File: VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetMessagesForConferenceQueryTests.cs (hunk 2 above)
Comment (jdssunil) [reply VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetMessagesForConferenceQueryTests.cs]: Removed

File: VideoAPI/VideoApi.Contract/Responses/MessageResponse.cs (hunk 1 above)
Comment (jdssunil) [reply VideoAPI/VideoApi.Contract/Responses/MessageResponse.cs]: Removed. This affects the unit tests. Please review the tests too.

File: VideoAPI/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs [hunk 7]
```diff
@@ -40,5 +44,17 @@ public void should_map_all_properties()
                 .Excluding(x => x.CurrentRoom)
             );
         }
+
+        bool ExcludeIdFromMessage1(IMemberInfo member)
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs]: please remove whichever method is unused

Comment (jdssunil) [reply VideoAPI/VideoApi.UnitTests/Mappings/ConferenceToDetailsResponseMapperTests.cs]: Done

**Pull request merged**